<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Martijn van Attekum" />

<meta name="date" content="2023-08-15" />

<title>Manipulating SummarizedExperiments with dplyr functions</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Manipulating SummarizedExperiments with
dplyr functions</h1>
<h4 class="author">Martijn van Attekum</h4>
<h4 class="date">2023-08-15</h4>



<p>The SummarizedExperiment (se) class offers a useful way to store
multiple row and column data along with the values from an experiment
and is widely used in computational biology. Although subsetting se’s is
possible with base R notation (ie using <code>[]</code>), se’s lack
methods for dplyr functions such as <code>filter</code> and
<code>select</code>, which excludes se’s from easily being used in
pipes. This package offers methods for the dplyr functions and will
automatically dispatch your se to the relevant function.</p>
<p>As se’s contain a data frame for both the rowData and colData, a
major difference using these functions in cleanse, is that we need to
<strong>specify whether we apply our function to the row or the
col</strong> of the se. Cleanse will then take care of updating the
se.</p>
<p>The package contains an example se called <code>seq_se</code>. The
example contains dummy data of expression and copy number values for 10
genes and 48 different conditions. To get an overview of the different
options available in the se, use <code>print_options</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(cleanse)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">data</span>(seq_se)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">print_options</span>(seq_se)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt; ** ASSAY OPTIONS FOR seq_se ** </span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&gt; expression, copy_number</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&gt; ** COLDATA OPTIONS FOR seq_se ** </span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt;   patient : 1, 2, 3, 4 </span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&gt;      site : skin, brain, liver </span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt;      time : 0, 4 </span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#&gt; treatment : A, B </span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co">#&gt; ** ROWDATA OPTIONS FOR seq_se ** </span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co">#&gt; gene_group : IL, NOTCH, TLR </span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#&gt;  gene_name : DLL1, DLL3, IL1R1, IL1R2, IL2RA, IL2RB, IL2RG, JAG1, TLR1, TLR2</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="co">#&gt; dim: 10 48 </span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="co">#&gt; assays(2): expression copy_number</span></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="co">#&gt; rownames: NULL</span></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="co">#&gt; rowData names(2): gene_group gene_name</span></span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="co">#&gt; colnames: NULL</span></span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a><span class="co">#&gt; colData names(4): patient site time treatment</span></span></code></pre></div>
<p>As you can see, seq_se contains expression and copy_number data for
samples taken from 3 different sites, which were treated with either
treatment A or B for 0 or 4 hours for 4 different patients. The genes
that were sequenced are from 3 different gene groups: IL, NOTCH, and
TLR. Each of the rowData / colData columns (eg patient, site,
gene_group) is refered to as a <strong>variable</strong> henceforth.</p>
<div id="using-dplyr-functions-to-subset-the-se" class="section level2">
<h2>Using dplyr functions to subset the se</h2>
<p>These functions either return a subset of the se or a rearranged se.
For each of them, the underlying assay values are updated accordingly,
so everything is kept in sync.</p>
<p>The <code>filter</code> function subsets rows / cols from the se
based on conditional filtering of a variable contained in either rowData
or colData respectively. Experiment values in the assay are dropped
concurrently with the update of the colData/rowData.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>genes_subset_se <span class="ot">&lt;-</span> seq_se <span class="sc">%&gt;%</span> <span class="fu">filter</span>(row, gene_group <span class="sc">==</span> <span class="st">&quot;IL&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">print_options</span>(genes_subset_se)  <span class="co">#note the change in available gene_groups</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt; ** ASSAY OPTIONS FOR genes_subset_se ** </span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#&gt; expression, copy_number</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#&gt; ** COLDATA OPTIONS FOR genes_subset_se ** </span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co">#&gt;   patient : 1, 2, 3, 4 </span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co">#&gt;      site : skin, brain, liver </span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt;      time : 0, 4 </span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#&gt; treatment : A, B </span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="co">#&gt; ** ROWDATA OPTIONS FOR genes_subset_se ** </span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co">#&gt; gene_group : IL </span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co">#&gt;  gene_name : IL1R1, IL1R2, IL2RA, IL2RB, IL2RG</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="co">#&gt; dim: 5 48 </span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a><span class="co">#&gt; assays(2): expression copy_number</span></span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a><span class="co">#&gt; rownames: NULL</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a><span class="co">#&gt; rowData names(2): gene_group gene_name</span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a><span class="co">#&gt; colnames: NULL</span></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a><span class="co">#&gt; colData names(4): patient site time treatment</span></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a><span class="fu">dim</span>(seq_se)</span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a><span class="co">#&gt; [1] 10 48</span></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a><span class="fu">dim</span>(genes_subset_se)</span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a><span class="co">#&gt; [1]  5 48</span></span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a></span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a>sample_subset_se <span class="ot">&lt;-</span> seq_se <span class="sc">%&gt;%</span> <span class="fu">filter</span>(col, treatment <span class="sc">==</span> <span class="st">&quot;B&quot;</span>, site <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;brain&quot;</span>, <span class="st">&quot;skin&quot;</span>))</span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a><span class="fu">print_options</span>(sample_subset_se) <span class="co">#note the change in available treatments and sites</span></span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a><span class="co">#&gt; ** ASSAY OPTIONS FOR sample_subset_se ** </span></span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a><span class="co">#&gt; expression, copy_number</span></span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a><span class="co">#&gt; ** COLDATA OPTIONS FOR sample_subset_se ** </span></span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a><span class="co">#&gt;   patient : 1, 2, 3, 4 </span></span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a><span class="co">#&gt;      site : skin, brain </span></span>
<span id="cb3-36"><a href="#cb3-36" tabindex="-1"></a><span class="co">#&gt;      time : 0, 4 </span></span>
<span id="cb3-37"><a href="#cb3-37" tabindex="-1"></a><span class="co">#&gt; treatment : B </span></span>
<span id="cb3-38"><a href="#cb3-38" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-39"><a href="#cb3-39" tabindex="-1"></a><span class="co">#&gt; ** ROWDATA OPTIONS FOR sample_subset_se ** </span></span>
<span id="cb3-40"><a href="#cb3-40" tabindex="-1"></a><span class="co">#&gt; gene_group : IL, NOTCH, TLR </span></span>
<span id="cb3-41"><a href="#cb3-41" tabindex="-1"></a><span class="co">#&gt;  gene_name : DLL1, DLL3, IL1R1, IL1R2, IL2RA, IL2RB, IL2RG, JAG1, TLR1, TLR2</span></span>
<span id="cb3-42"><a href="#cb3-42" tabindex="-1"></a><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span id="cb3-43"><a href="#cb3-43" tabindex="-1"></a><span class="co">#&gt; dim: 10 16 </span></span>
<span id="cb3-44"><a href="#cb3-44" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb3-45"><a href="#cb3-45" tabindex="-1"></a><span class="co">#&gt; assays(2): expression copy_number</span></span>
<span id="cb3-46"><a href="#cb3-46" tabindex="-1"></a><span class="co">#&gt; rownames: NULL</span></span>
<span id="cb3-47"><a href="#cb3-47" tabindex="-1"></a><span class="co">#&gt; rowData names(2): gene_group gene_name</span></span>
<span id="cb3-48"><a href="#cb3-48" tabindex="-1"></a><span class="co">#&gt; colnames: NULL</span></span>
<span id="cb3-49"><a href="#cb3-49" tabindex="-1"></a><span class="co">#&gt; colData names(4): patient site time treatment</span></span>
<span id="cb3-50"><a href="#cb3-50" tabindex="-1"></a><span class="fu">dim</span>(seq_se)</span>
<span id="cb3-51"><a href="#cb3-51" tabindex="-1"></a><span class="co">#&gt; [1] 10 48</span></span>
<span id="cb3-52"><a href="#cb3-52" tabindex="-1"></a><span class="fu">dim</span>(sample_subset_se)</span>
<span id="cb3-53"><a href="#cb3-53" tabindex="-1"></a><span class="co">#&gt; [1] 10 16</span></span></code></pre></div>
<p>To subset a se by position, <code>slice</code> can be used:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>seq_se <span class="sc">%&gt;%</span> cleanse<span class="sc">::</span><span class="fu">slice</span>(col, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>) <span class="co">#select the first 10 columns</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#&gt; dim: 10 10 </span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt; assays(2): expression copy_number</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#&gt; rownames: NULL</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt; rowData names(2): gene_group gene_name</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt; colnames: NULL</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt; colData names(4): patient site time treatment</span></span></code></pre></div>
<p><code>arrange</code> is used to consecutively sort a data frame by a
variable. In this case, we arrange the rows, first by gene_name and next
by gene_group.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>seq_se_reordered <span class="ot">&lt;-</span> seq_se <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(row, gene_name, gene_group)</span></code></pre></div>
<p>The slice_sample() function behaves similar to dplyr’s equivalent by
selecting random rows or cols:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">slice_sample</span>(seq_se, col, <span class="at">n=</span><span class="dv">3</span>)<span class="co">#note the change in dim</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#&gt; dim: 10 3 </span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#&gt; assays(2): expression copy_number</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#&gt; rownames: NULL</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt; rowData names(2): gene_group gene_name</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; colnames: NULL</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; colData names(4): patient site time treatment</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="fu">slice_sample</span>(seq_se, row, <span class="at">prop=</span>.<span class="dv">2</span>) <span class="co">#note the change in dim</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="co">#&gt; dim: 2 48 </span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="co">#&gt; assays(2): expression copy_number</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co">#&gt; rownames: NULL</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="co">#&gt; rowData names(2): gene_group gene_name</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="co">#&gt; colnames: NULL</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="co">#&gt; colData names(4): patient site time treatment</span></span></code></pre></div>
</div>
<div id="using-dplyr-functions-to-change-the-ses-metadata" class="section level2">
<h2>Using dplyr functions to change the se’s metadata</h2>
<p>These functions return a se with updated rowData or colData. The
dimensions and assay values of the se will not be changed by these
functions.</p>
<p><code>select</code> can be used to select variables, and
<code>rename</code> will rename these variables.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># remove the time variable after filtering for time == 0</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>seq_se_min_time <span class="ot">&lt;-</span> seq_se <span class="sc">%&gt;%</span> cleanse<span class="sc">::</span><span class="fu">filter</span>(col, time <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> cleanse<span class="sc">::</span><span class="fu">select</span>(col, <span class="sc">-</span>time)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="fu">print_options</span>(seq_se_min_time)  <span class="co">#note the time variable has disappeared from the colData</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#&gt; ** ASSAY OPTIONS FOR seq_se_min_time ** </span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#&gt; expression, copy_number</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co">#&gt; ** COLDATA OPTIONS FOR seq_se_min_time ** </span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="co">#&gt;   patient : 1, 2, 3, 4 </span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="co">#&gt;      site : skin, brain, liver </span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="co">#&gt; treatment : A, B </span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a><span class="co">#&gt; ** ROWDATA OPTIONS FOR seq_se_min_time ** </span></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a><span class="co">#&gt; gene_group : IL, NOTCH, TLR </span></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a><span class="co">#&gt;  gene_name : DLL1, DLL3, IL1R1, IL1R2, IL2RA, IL2RB, IL2RG, JAG1, TLR1, TLR2</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a><span class="co">#&gt; dim: 10 24 </span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a><span class="co">#&gt; assays(2): expression copy_number</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a><span class="co">#&gt; rownames: NULL</span></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a><span class="co">#&gt; rowData names(2): gene_group gene_name</span></span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a><span class="co">#&gt; colnames: NULL</span></span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a><span class="co">#&gt; colData names(3): patient site treatment</span></span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a></span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a><span class="co"># rename the time variable after changing it to minutes</span></span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a>seq_se_ren_time <span class="ot">&lt;-</span> seq_se <span class="sc">%&gt;%</span> cleanse<span class="sc">::</span><span class="fu">mutate</span>(col, <span class="at">time =</span> (time <span class="sc">*</span> <span class="dv">60</span>)) <span class="sc">%&gt;%</span> cleanse<span class="sc">::</span><span class="fu">rename</span>(col, <span class="at">time_mins =</span> time)</span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a><span class="fu">print_options</span>(seq_se_ren_time)  <span class="co">#note the time variable is now called time_mins</span></span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a><span class="co">#&gt; ** ASSAY OPTIONS FOR seq_se_ren_time ** </span></span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a><span class="co">#&gt; expression, copy_number</span></span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-30"><a href="#cb7-30" tabindex="-1"></a><span class="co">#&gt; ** COLDATA OPTIONS FOR seq_se_ren_time ** </span></span>
<span id="cb7-31"><a href="#cb7-31" tabindex="-1"></a><span class="co">#&gt;   patient : 1, 2, 3, 4 </span></span>
<span id="cb7-32"><a href="#cb7-32" tabindex="-1"></a><span class="co">#&gt;      site : skin, brain, liver </span></span>
<span id="cb7-33"><a href="#cb7-33" tabindex="-1"></a><span class="co">#&gt; time_mins : 0, 240 </span></span>
<span id="cb7-34"><a href="#cb7-34" tabindex="-1"></a><span class="co">#&gt; treatment : A, B </span></span>
<span id="cb7-35"><a href="#cb7-35" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-36"><a href="#cb7-36" tabindex="-1"></a><span class="co">#&gt; ** ROWDATA OPTIONS FOR seq_se_ren_time ** </span></span>
<span id="cb7-37"><a href="#cb7-37" tabindex="-1"></a><span class="co">#&gt; gene_group : IL, NOTCH, TLR </span></span>
<span id="cb7-38"><a href="#cb7-38" tabindex="-1"></a><span class="co">#&gt;  gene_name : DLL1, DLL3, IL1R1, IL1R2, IL2RA, IL2RB, IL2RG, JAG1, TLR1, TLR2</span></span>
<span id="cb7-39"><a href="#cb7-39" tabindex="-1"></a><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span id="cb7-40"><a href="#cb7-40" tabindex="-1"></a><span class="co">#&gt; dim: 10 48 </span></span>
<span id="cb7-41"><a href="#cb7-41" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb7-42"><a href="#cb7-42" tabindex="-1"></a><span class="co">#&gt; assays(2): expression copy_number</span></span>
<span id="cb7-43"><a href="#cb7-43" tabindex="-1"></a><span class="co">#&gt; rownames: NULL</span></span>
<span id="cb7-44"><a href="#cb7-44" tabindex="-1"></a><span class="co">#&gt; rowData names(2): gene_group gene_name</span></span>
<span id="cb7-45"><a href="#cb7-45" tabindex="-1"></a><span class="co">#&gt; colnames: NULL</span></span>
<span id="cb7-46"><a href="#cb7-46" tabindex="-1"></a><span class="co">#&gt; colData names(4): patient site time_mins treatment</span></span></code></pre></div>
<p><code>mutate</code> adds or changes variables. If we for instance
want to change the time for the samples from hours to minutes, we can
do</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>seq_se_mins <span class="ot">&lt;-</span> seq_se <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(col, <span class="at">time =</span> (time <span class="sc">*</span> <span class="dv">60</span>))</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>seq_se<span class="sc">$</span>time</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co">#&gt;  [1] 0 0 0 0 0 0 0 0 0 0 0 0 4 4 4 4 4 4 4 4 4 4 4 4 0 0 0 0 0 0 0 0 0 0 0 0 4 4</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="co">#&gt; [39] 4 4 4 4 4 4 4 4 4 4</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>seq_se_mins<span class="sc">$</span>time</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co">#&gt;  [1]   0   0   0   0   0   0   0   0   0   0   0   0 240 240 240 240 240 240 240</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="co">#&gt; [20] 240 240 240 240 240   0   0   0   0   0   0   0   0   0   0   0   0 240 240</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co">#&gt; [39] 240 240 240 240 240 240 240 240 240 240</span></span></code></pre></div>
<p>or to combine the gene groups and gene names to one new variable in
the rowData</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>seq_se_gene_comb <span class="ot">&lt;-</span> seq_se <span class="sc">%&gt;%</span> </span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="fu">mutate</span>(row, <span class="at">group_and_name =</span> <span class="fu">paste</span>(gene_group, gene_name, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>))</span></code></pre></div>
<p>A non-dplyr function that will change the metadata is
<code>drop_metadata</code>: This function will drop all rowData and
colData variables that have only 1 value. Typically used after
subsetting:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>seq_se_dropped <span class="ot">&lt;-</span> seq_se <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="fu">filter</span>(col, time <span class="sc">==</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>  <span class="fu">drop_metadata</span>()</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="fu">print_options</span>(seq_se_dropped) <span class="co">#note the time variable from colData is dropped as all values were 4</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="co">#&gt; ** ASSAY OPTIONS FOR seq_se_dropped ** </span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a><span class="co">#&gt; expression, copy_number</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a><span class="co">#&gt; ** COLDATA OPTIONS FOR seq_se_dropped ** </span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="co">#&gt;   patient : 1, 2, 3, 4 </span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="co">#&gt;      site : skin, brain, liver </span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a><span class="co">#&gt; treatment : A, B </span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a><span class="co">#&gt; ** ROWDATA OPTIONS FOR seq_se_dropped ** </span></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a><span class="co">#&gt; gene_group : IL, NOTCH, TLR </span></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a><span class="co">#&gt;  gene_name : DLL1, DLL3, IL1R1, IL1R2, IL2RA, IL2RB, IL2RG, JAG1, TLR1, TLR2</span></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a><span class="co">#&gt; dim: 10 24 </span></span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a><span class="co">#&gt; metadata(0):</span></span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a><span class="co">#&gt; assays(2): expression copy_number</span></span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a><span class="co">#&gt; rownames: NULL</span></span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a><span class="co">#&gt; rowData names(2): gene_group gene_name</span></span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a><span class="co">#&gt; colnames: NULL</span></span>
<span id="cb10-23"><a href="#cb10-23" tabindex="-1"></a><span class="co">#&gt; colData names(3): patient site treatment</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="co">#&gt; R version 4.3.1 (2023-06-16)</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="co">#&gt; Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co">#&gt; Running under: macOS Ventura 13.4.1</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co">#&gt; locale:</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="co">#&gt; [1] C/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a><span class="co">#&gt; time zone: Europe/Berlin</span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a><span class="co">#&gt; tzcode source: internal</span></span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb11-20"><a href="#cb11-20" tabindex="-1"></a><span class="co">#&gt; [1] cleanse_0.99.0 dplyr_1.1.2   </span></span>
<span id="cb11-21"><a href="#cb11-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb11-22"><a href="#cb11-22" tabindex="-1"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb11-23"><a href="#cb11-23" tabindex="-1"></a><span class="co">#&gt;  [1] sass_0.4.7                  utf8_1.2.3                 </span></span>
<span id="cb11-24"><a href="#cb11-24" tabindex="-1"></a><span class="co">#&gt;  [3] generics_0.1.3              tidyr_1.3.0                </span></span>
<span id="cb11-25"><a href="#cb11-25" tabindex="-1"></a><span class="co">#&gt;  [5] SparseArray_1.1.11          bitops_1.0-7               </span></span>
<span id="cb11-26"><a href="#cb11-26" tabindex="-1"></a><span class="co">#&gt;  [7] xml2_1.3.5                  stringi_1.7.12             </span></span>
<span id="cb11-27"><a href="#cb11-27" tabindex="-1"></a><span class="co">#&gt;  [9] lattice_0.21-8              digest_0.6.33              </span></span>
<span id="cb11-28"><a href="#cb11-28" tabindex="-1"></a><span class="co">#&gt; [11] hms_1.1.3                   magrittr_2.0.3             </span></span>
<span id="cb11-29"><a href="#cb11-29" tabindex="-1"></a><span class="co">#&gt; [13] evaluate_0.21               grid_4.3.1                 </span></span>
<span id="cb11-30"><a href="#cb11-30" tabindex="-1"></a><span class="co">#&gt; [15] fastmap_1.1.1               pkgload_1.3.2.1            </span></span>
<span id="cb11-31"><a href="#cb11-31" tabindex="-1"></a><span class="co">#&gt; [17] jsonlite_1.8.7              rprojroot_2.0.3            </span></span>
<span id="cb11-32"><a href="#cb11-32" tabindex="-1"></a><span class="co">#&gt; [19] Matrix_1.6-1                GenomeInfoDb_1.37.2        </span></span>
<span id="cb11-33"><a href="#cb11-33" tabindex="-1"></a><span class="co">#&gt; [21] purrr_1.0.2                 fansi_1.0.4                </span></span>
<span id="cb11-34"><a href="#cb11-34" tabindex="-1"></a><span class="co">#&gt; [23] jquerylib_0.1.4             abind_1.4-5                </span></span>
<span id="cb11-35"><a href="#cb11-35" tabindex="-1"></a><span class="co">#&gt; [25] cli_3.6.1                   rlang_1.1.1                </span></span>
<span id="cb11-36"><a href="#cb11-36" tabindex="-1"></a><span class="co">#&gt; [27] crayon_1.5.2                XVector_0.41.1             </span></span>
<span id="cb11-37"><a href="#cb11-37" tabindex="-1"></a><span class="co">#&gt; [29] Biobase_2.61.0              cachem_1.0.8               </span></span>
<span id="cb11-38"><a href="#cb11-38" tabindex="-1"></a><span class="co">#&gt; [31] yaml_2.3.7                  withr_2.5.0                </span></span>
<span id="cb11-39"><a href="#cb11-39" tabindex="-1"></a><span class="co">#&gt; [33] DelayedArray_0.27.10        S4Arrays_1.1.5             </span></span>
<span id="cb11-40"><a href="#cb11-40" tabindex="-1"></a><span class="co">#&gt; [35] tools_4.3.1                 tzdb_0.4.0                 </span></span>
<span id="cb11-41"><a href="#cb11-41" tabindex="-1"></a><span class="co">#&gt; [37] GenomeInfoDbData_1.2.10     SummarizedExperiment_1.31.1</span></span>
<span id="cb11-42"><a href="#cb11-42" tabindex="-1"></a><span class="co">#&gt; [39] BiocGenerics_0.47.0         vctrs_0.6.3                </span></span>
<span id="cb11-43"><a href="#cb11-43" tabindex="-1"></a><span class="co">#&gt; [41] R6_2.5.1                    matrixStats_1.0.0          </span></span>
<span id="cb11-44"><a href="#cb11-44" tabindex="-1"></a><span class="co">#&gt; [43] stats4_4.3.1                lifecycle_1.0.3            </span></span>
<span id="cb11-45"><a href="#cb11-45" tabindex="-1"></a><span class="co">#&gt; [45] zlibbioc_1.47.0             stringr_1.5.0              </span></span>
<span id="cb11-46"><a href="#cb11-46" tabindex="-1"></a><span class="co">#&gt; [47] S4Vectors_0.39.1            IRanges_2.35.2             </span></span>
<span id="cb11-47"><a href="#cb11-47" tabindex="-1"></a><span class="co">#&gt; [49] pkgconfig_2.0.3             desc_1.4.2                 </span></span>
<span id="cb11-48"><a href="#cb11-48" tabindex="-1"></a><span class="co">#&gt; [51] bslib_0.5.1                 pillar_1.9.0               </span></span>
<span id="cb11-49"><a href="#cb11-49" tabindex="-1"></a><span class="co">#&gt; [53] glue_1.6.2                  xfun_0.40                  </span></span>
<span id="cb11-50"><a href="#cb11-50" tabindex="-1"></a><span class="co">#&gt; [55] tibble_3.2.1                GenomicRanges_1.53.1       </span></span>
<span id="cb11-51"><a href="#cb11-51" tabindex="-1"></a><span class="co">#&gt; [57] tidyselect_1.2.0            rstudioapi_0.15.0          </span></span>
<span id="cb11-52"><a href="#cb11-52" tabindex="-1"></a><span class="co">#&gt; [59] MatrixGenerics_1.13.1       knitr_1.43                 </span></span>
<span id="cb11-53"><a href="#cb11-53" tabindex="-1"></a><span class="co">#&gt; [61] htmltools_0.5.6             rmarkdown_2.24             </span></span>
<span id="cb11-54"><a href="#cb11-54" tabindex="-1"></a><span class="co">#&gt; [63] readr_2.1.4                 compiler_4.3.1             </span></span>
<span id="cb11-55"><a href="#cb11-55" tabindex="-1"></a><span class="co">#&gt; [65] roxygen2_7.2.3              RCurl_1.98-1.12</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
